index.bs

```diff
@@ -8465,11 +8465,24 @@ IDL [=promise type=] values are represented by ECMAScript [=PromiseCapability=]
         1.  [=Reject=] |promise| with |reason|.
     1.  [=Wait for all=] with |promises|, given |successSteps| and |failureSteps|.
     1.  Return |promise|.
+
+    <p class="note">This definition is useful when you wish to aggregate the results of multiple
+    promises, and then produce another promise from them, in the same way that
+    {{Promise/all()|Promise.all()}} functions for JavaScript code. If you do not need to produce
+    another promise, then [=waiting for all=] is likely better.
 </div>
 
-This phrase is useful when you wish to aggregate the results of multiple promises, and then produce
-another promise from them, in the same way that {{Promise/all()|Promise.all()}} functions for
-JavaScript code.
+<div algorithm>
+    To <dfn export lt="mark a promise as handled|mark as handled">mark as handled</dfn> a
+    <code><a interface>Promise</a>&lt;<var ignore>T</var>&gt;</code> |promise|, set
+    |promise|.\[[Promise]].\[[PromiseIsHandled]] to true.
+
+    <p class="note">This definition is useful for promises which you expect rejections to often be
+    ignored; it ensures such promises do not cause {{Window/unhandledrejection}} events. The most
+    common use case is for promise properties, which the web developer might or might not consult.
+    An example is the {{WritableStreamDefaultWriter/closed|writableStreamWriter.closed}} promise,
+    which is [=marked as handled=] immediately after it is [=rejected=].
```

Comment:
Would it make sense to offer an operation that rejects and marks as handled as a combined operation? Or are there cases where they need to be disjoint?

Reply:
So what I've found recently (in app history) is that it's actually simplest to just mark the promise as handled upon creation. What streams does is perhaps unnecessarily complicated, and I wonder if maybe I should remove this example or make it more vague.

The advantage of this is that you can centralize your "this promise is special" logic at creation time, instead of all the different places you reject it. So if anything maybe a "new already-handled promise" algorithm would be useful? But for now my instinct is to just expose the primitive, since so far use cases are pretty rare.

Reply:
For now I just removed "which is marked as handled immediately after it is rejected" to make it a bit more vague.